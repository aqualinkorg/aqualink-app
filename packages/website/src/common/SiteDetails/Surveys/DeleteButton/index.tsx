import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import {
  withStyles,
  WithStyles,
  createStyles,
  IconButton,
  Collapse,
  LinearProgress,
} from "@material-ui/core";
import Alert from "@material-ui/lab/Alert";
import DeleteOutlineIcon from "@material-ui/icons/DeleteOutline";
import CloseIcon from "@material-ui/icons/Close";
import moment from "moment";

import surveyServices from "../../../../services/surveyServices";
import { userInfoSelector } from "../../../../store/User/userSlice";
import { surveysRequest } from "../../../../store/Survey/surveyListSlice";
import DeleteDialog, { Action } from "../../../Dialog";

const DeleteButton = ({
  reefId,
  surveyId,
  diveDate,
  classes,
}: DeleteButtonProps) => {
  const user = useSelector(userInfoSelector);
  const dispatch = useDispatch();
  const [open, setOpen] = useState<boolean>(false);
  const [alertOpen, setAlertOpen] = useState<boolean>(false);
  const [alertText, setAlertText] = useState<string>("");
  const [loading, setLoading] = useState<boolean>(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
    setAlertOpen(false);
    setAlertText("");
  };

  const onSurveyDelete = () => {
    if (surveyId && user && user.token) {
      setLoading(true);
      surveyServices
        .deleteSurvey(reefId, surveyId, user.token)
        .then(() => {
          dispatch(surveysRequest(`${reefId}`));
        })
        .catch((error) => {
          setAlertOpen(true);
          setAlertText(error.message);
        })
        .finally(() => setLoading(false));
    }
  };

  const dialogActions: Action[] = [
    {
      size: "small",
      variant: "contained",
      color: "secondary",
      text: "No",
      action: handleClose,
    },
    {
      size: "small",
      variant: "contained",
      color: "primary",
      text: "Yes",
      action: onSurveyDelete,
    },
  ];

  return (
    <>
      <IconButton onClick={handleClickOpen}>
        <DeleteOutlineIcon color="primary" />
      </IconButton>
      <DeleteDialog
        open={open}
        onClose={handleClose}
        header={`Are you sure you would like to delete the survey for ${moment(
          diveDate
        ).format(
          "MM/DD/YYYY"
        )}? It will delete all media assosciated with this survey.`}
        content={
          <>
            {loading && <LinearProgress />}
            <Collapse className={classes.alert} in={alertOpen}>
              <Alert
                severity="error"
                action={
                  <IconButton
                    color="inherit"
                    size="small"
                    onClick={() => {
                      setAlertOpen(false);
                    }}
                  >
                    <CloseIcon fontSize="inherit" />
                  </IconButton>
                }
              >
                {alertText}
              </Alert>
            </Collapse>
          </>
        }
        actions={dialogActions}
      />
    </>
  );
};

const styles = () =>
  createStyles({
    alert: {
      width: "100%",
    },
  });

interface DeleteButtonIncomingProps {
  surveyId?: number | null;
  diveDate?: string | null;
  reefId: number;
}

DeleteButton.defaultProps = {
  surveyId: null,
  diveDate: null,
};

type DeleteButtonProps = DeleteButtonIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(DeleteButton);
