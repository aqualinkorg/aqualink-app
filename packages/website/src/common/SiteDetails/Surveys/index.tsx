import React, { useState, useEffect, ChangeEvent } from 'react';
import {
  useTheme,
  useMediaQuery,
  Theme,
  Grid,
  Typography,
  Select,
  FormControl,
  MenuItem,
  Box,
  SelectChangeEvent,
} from '@mui/material';
import createStyles from '@mui/styles/createStyles';
import makeStyles from '@mui/styles/makeStyles';
import { useDispatch, useSelector } from 'react-redux';

import observationOptions from 'constants/uploadDropdowns';
import { setSiteSurveyPoints } from 'store/Sites/selectedSiteSlice';
import { userInfoSelector } from 'store/User/userSlice';
import {
  surveysRequest,
  updateSurveyPointName,
} from 'store/Survey/surveyListSlice';
import { setSelectedPoi } from 'store/Survey/surveySlice';
import { SurveyMedia } from 'store/Survey/types';
import { Site } from 'store/Sites/types';
import { useBodyLength } from 'hooks/useBodyLength';
import { isAdmin } from 'helpers/user';
import { getAxiosErrorMessage } from 'helpers/errors';
import siteServices from 'services/siteServices';
import surveyServices from 'services/surveyServices';
import PointSelector from './PointSelector';
import Timeline from './Timeline';
import DeleteSurveyPointDialog, { Action } from '../../Dialog';

const Surveys = ({ site }: SurveysProps) => {
  const classes = useStyles();
  const theme = useTheme();
  const isTablet = useMediaQuery(theme.breakpoints.down('lg'));
  const isLoading = !site;
  const [point, setPoint] = useState<string>('All');
  const pointOptions = site?.surveyPoints || [];
  const [deleteSurveyPointDialogOpen, setDeleteSurveyPointDialogOpen] =
    useState<boolean>(false);
  const [editSurveyPointNameDraft, seteditSurveyPointNameDraft] = useState<
    string | null
  >();
  const [editSurveyPointNameLoading, seteditSurveyPointNameLoading] =
    useState<boolean>(false);
  const [surveyPointToDelete, setSurveyPointToDelete] = useState<number | null>(
    null,
  );
  const [isDeletePointLoading, setIsDeletePointLoading] = useState(false);
  const [deletePointError, setDeletePointError] = useState<string>();
  const [observation, setObservation] = useState<
    SurveyMedia['observations'] | 'any'
  >('any');
  const user = useSelector(userInfoSelector);
  const isSiteAdmin = site ? isAdmin(user, site.id) : false;
  const dispatch = useDispatch();

  const bodyLength = useBodyLength();

  const surveyPointToDeleteName = pointOptions.find(
    ({ id }) => id === surveyPointToDelete,
  )?.name;

  useEffect(() => {
    dispatch(setSelectedPoi(point));
  }, [dispatch, point]);

  const onDeleteSurveyPointButtonClick = (id: number) => {
    setDeleteSurveyPointDialogOpen(true);
    setSurveyPointToDelete(id);
  };

  const handlePointChange = (event: SelectChangeEvent<unknown>) => {
    setPoint(event.target.value as string);
  };

  const handleObservationChange = (event: SelectChangeEvent<unknown>) => {
    setObservation(event.target.value as SurveyMedia['observations'] | 'any');
  };

  const pointIdFinder = (name: string) => {
    return pointOptions.find((option) => option.name === name)?.id || -1;
  };

  const handleDeleteSurveyPointDialogClose = () => {
    setDeleteSurveyPointDialogOpen(false);
    setSurveyPointToDelete(null);
    setIsDeletePointLoading(false);
    setDeletePointError(undefined);
  };

  const handleSurveyPointDelete = async () => {
    if (site && typeof surveyPointToDelete === 'number') {
      setIsDeletePointLoading(true);
      try {
        await siteServices.deleteSiteSurveyPoint(
          surveyPointToDelete,
          user?.token,
        );

        dispatch(
          setSiteSurveyPoints(
            pointOptions.filter((option) => option.id !== surveyPointToDelete),
          ),
        );
        dispatch(surveysRequest(`${site.id}`));
        setDeleteSurveyPointDialogOpen(false);
        setSurveyPointToDelete(null);
      } catch (error) {
        setDeletePointError(getAxiosErrorMessage(error));
      } finally {
        setIsDeletePointLoading(false);
      }
    }
  };

  const enableeditSurveyPointName = (id: number) => {
    const initialName = pointOptions.find((item) => item.id === id)?.name;
    seteditSurveyPointNameDraft(initialName);
  };

  const disableeditSurveyPointName = () =>
    seteditSurveyPointNameDraft(undefined);

  const onChangeSurveyPointName = (
    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
  ) => seteditSurveyPointNameDraft(event.target.value);

  const submitSurveyPointNameUpdate = (key: number) => {
    const newName = editSurveyPointNameDraft;
    if (newName && user?.token) {
      seteditSurveyPointNameLoading(true);
      surveyServices
        .updatePoi(key, { name: newName }, user.token)
        .then(() => {
          // Update point name for featured image card
          dispatch(updateSurveyPointName({ id: key, name: newName }));

          // If the updated point was previously selected, update its value
          const prevName = pointOptions.find((item) => item.id === key)?.name;
          if (prevName === point) {
            setPoint(newName);
          }

          // Update point options
          dispatch(
            setSiteSurveyPoints(
              pointOptions.map((item) => {
                if (item.id === key) {
                  return {
                    ...item,
                    name: newName,
                  };
                }
                return item;
              }),
            ),
          );
          seteditSurveyPointNameDraft(undefined);
        })
        .catch(console.error)
        .finally(() => seteditSurveyPointNameLoading(false));
    }
  };

  const deleteSurveyPointDialogActions: Action[] = [
    {
      size: 'small',
      variant: 'contained',
      color: 'secondary',
      text: 'No',
      disabled: isDeletePointLoading,
      action: handleDeleteSurveyPointDialogClose,
    },
    {
      size: 'small',
      variant: 'contained',
      color: 'primary',
      text: 'Yes',
      loading: isDeletePointLoading,
      disabled: isDeletePointLoading,
      action: handleSurveyPointDelete,
    },
  ];

  return (
    <>
      <DeleteSurveyPointDialog
        open={deleteSurveyPointDialogOpen}
        onClose={handleDeleteSurveyPointDialogClose}
        error={deletePointError}
        header={`Delete ${surveyPointToDeleteName}`}
        content={
          <Typography color="textSecondary">
            Are you sure you want to delete this survey point? It will be
            deleted across all surveys.
          </Typography>
        }
        actions={deleteSurveyPointDialogActions}
      />
      <Grid
        className={classes.root}
        container
        justifyContent="center"
        spacing={2}
      >
        <Box
          id="surveys"
          bgcolor="#f5f6f6"
          position="absolute"
          height="100%"
          width={bodyLength}
          zIndex="-1"
        />
        {!isLoading && (
          <Grid
            className={classes.surveyWrapper}
            container
            justifyContent="space-between"
            item
            lg={12}
            xs={12}
            alignItems="baseline"
            spacing={isTablet ? 4 : 1}
          >
            <Grid
              container
              justifyContent={isTablet ? 'flex-start' : 'center'}
              item
              md={12}
              lg={3}
            >
              <Typography className={classes.title}>Survey History</Typography>
            </Grid>
            <PointSelector
              siteId={site?.id}
              pointOptions={pointOptions}
              point={point}
              pointId={pointIdFinder(point)}
              editSurveyPointNameDraft={editSurveyPointNameDraft}
              isSiteAdmin={isSiteAdmin}
              editSurveyPointNameLoading={editSurveyPointNameLoading}
              onChangeSurveyPointName={onChangeSurveyPointName}
              handlePointChange={handlePointChange}
              enableeditSurveyPointName={enableeditSurveyPointName}
              disableeditSurveyPointName={disableeditSurveyPointName}
              submitSurveyPointNameUpdate={submitSurveyPointNameUpdate}
              onDeleteButtonClick={onDeleteSurveyPointButtonClick}
            />
            <Grid
              container
              alignItems="center"
              justifyContent={isTablet ? 'flex-start' : 'center'}
              item
              md={12}
              lg={4}
              spacing={1}
            >
              {/* TODO - Make observation a required field. */}
              <Grid item>
                <Typography variant="h6" className={classes.subTitle}>
                  Observation:
                </Typography>
              </Grid>
              <Grid item className={classes.selectorWrapper}>
                <FormControl variant="standard" className={classes.formControl}>
                  <Select
                    variant="standard"
                    labelId="survey-observation"
                    id="survey-observation"
                    name="survey-observation"
                    value={observation}
                    onChange={handleObservationChange}
                    className={classes.selectedItem}
                    inputProps={{ className: classes.textField }}
                  >
                    <MenuItem value="any">
                      <Typography className={classes.menuItem} variant="h6">
                        Any
                      </Typography>
                    </MenuItem>
                    {observationOptions.map((item) => (
                      <MenuItem
                        className={classes.menuItem}
                        value={item.key}
                        key={item.key}
                        title={item.value}
                      >
                        {item.value}
                      </MenuItem>
                    ))}
                  </Select>
                </FormControl>
              </Grid>
            </Grid>
          </Grid>
        )}
        <Grid container justifyContent="center" item xs={12} lg={12}>
          <Timeline
            isAdmin={isSiteAdmin}
            loading={isLoading}
            addNewButton
            siteId={site?.id}
            timeZone={site?.timezone}
            observation={observation}
            pointName={point}
            pointId={pointIdFinder(point)}
          />
        </Grid>
      </Grid>
    </>
  );
};

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      marginTop: '5rem',
      position: 'relative',
    },
    surveyWrapper: {
      marginTop: '5rem',
    },
    title: {
      fontSize: 22,
      lineHeight: 1.45,
      color: '#2a2a2a',
      marginBottom: '1rem',
    },
    subTitle: {
      lineHeight: 1,
      color: '#474747',
      marginRight: '1rem',
    },
    selectorWrapper: {
      [theme.breakpoints.down('sm')]: {
        width: '100%',
      },
    },
    formControl: {
      minWidth: 120,
      maxWidth: 240,
    },
    selectedItem: {
      color: theme.palette.primary.main,
    },
    menuItem: {
      color: theme.palette.primary.main,
      width: '100%',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      display: 'block',
    },
    textField: {
      width: '100%',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      display: 'block',
    },
  }),
);

interface SurveysProps {
  site?: Site;
}

export default Surveys;
