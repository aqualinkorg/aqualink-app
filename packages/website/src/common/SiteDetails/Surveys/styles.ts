import theme from "../../../layout/App/theme";

const styles = {
  dateWrapper: {
    marginBottom: "1rem",
  },
  dates: {
    fontWeight: 500,
    lineHeight: 0.81,
    color: "#757575",
  },
  addNewButton: {
    color: "#979797",
    height: "2rem",
    width: "2rem",
  },
  surveyCardWrapper: {
    marginBottom: "2rem",
    [theme.breakpoints.up("lg")]: {
      maxWidth: "65%",
    },
  },
  surveyCard: {
    width: "100%",
    backgroundColor: theme.palette.primary.light,
    border: 1,
    borderStyle: "solid",
    borderColor: "#dddddd",
    borderRadius: 2,
    height: 320,
    [theme.breakpoints.down("sm")]: {
      height: 640,
    },
    [theme.breakpoints.down("xs")]: {
      width: "100%",
    },
  },
  cardImage: {
    height: "100%",
    width: "100%",
  },
  cardFields: {
    fontWeight: 500,
    lineHeight: 2,
    color: "#9ea6aa",
  },
  cardValues: {
    lineHeight: 2,
    color: "#2f2f2f",
  },
};

export default styles;
