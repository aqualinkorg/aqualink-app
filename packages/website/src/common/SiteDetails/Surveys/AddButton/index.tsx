import React from 'react';
import { Link } from 'react-router-dom';
import { Button, Typography } from '@mui/material';
import { WithStyles } from '@mui/styles';
import withStyles from '@mui/styles/withStyles';
import createStyles from '@mui/styles/createStyles';
import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';
import { grey } from '@mui/material/colors';

const AddButton = ({ siteId, classes }: AddButtonProps) => {
  return (
    <Button
      className={classes.addSurveyButton}
      startIcon={<AddCircleOutlineIcon />}
      component={Link}
      to={`/sites/${siteId}/new_survey`}
      classes={{ iconSizeMedium: classes.addSurveyButtonIcon }}
    >
      <Typography color="inherit" variant="h6">
        ADD NEW SURVEY
      </Typography>
    </Button>
  );
};

const styles = () =>
  createStyles({
    addSurveyButton: {
      color: grey[500],
      '&:hover': {
        color: grey[500],
      },
    },
    addSurveyButtonIcon: {
      '& > *:first-child': {
        fontSize: 32,
      },
    },
  });

interface AddButtonIncomingProps {
  siteId: number;
}

type AddButtonProps = AddButtonIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(AddButton);
