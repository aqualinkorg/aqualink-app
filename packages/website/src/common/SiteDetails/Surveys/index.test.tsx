import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";

import Surveys from ".";
import { mockUser } from "../../../mocks/mockUser";
import { mockSurvey } from "../../../mocks/mockSurvey";
import { mockReef } from "../../../mocks/mockReef";

const mockStore = configureStore([]);

describe("Surveys", () => {
  let element: HTMLElement;
  beforeEach(() => {
    const store = mockStore({
      surveyList: {
        list: [mockSurvey],
        loading: false,
        error: null,
      },
      user: {
        userInfo: mockUser,
        loading: false,
        error: null,
      },
    });

    store.dispatch = jest.fn();

    element = render(
      <Provider store={store}>
        <Router>
          <Surveys reef={mockReef} />
        </Router>
      </Provider>
    ).container;
  });

  it("should render with given state from Redux store", () => {
    expect(element).toMatchSnapshot();
  });
});
