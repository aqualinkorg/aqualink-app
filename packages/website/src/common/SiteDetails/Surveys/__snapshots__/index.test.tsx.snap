// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surveys should render with given state from Redux store 1`] = `
<div>
  <mock-dialog
    fullwidth="true"
    open="false"
  >
    <mock-dialogtitle
      classname="Dialog-dialogTitle-10"
    >
      <mock-typography>
        Are you sure you want to delete this survey point? It will be deleted across all surveys.
      </mock-typography>
    </mock-dialogtitle>
    <mock-dialogcontent />
    <mock-dialogactions>
      <mock-button
        color="secondary"
        size="small"
        variant="contained"
      >
        No
      </mock-button>
      <mock-button
        color="primary"
        size="small"
        variant="contained"
      >
        Yes
      </mock-button>
    </mock-dialogactions>
  </mock-dialog>
  <div
    class="MuiGrid-root Surveys-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
  >
    <mock-box
      bgcolor="#f5f6f6"
      height="100%"
      position="absolute"
      zindex="-1"
    />
    <div
      class="MuiGrid-root Surveys-surveyWrapper-2 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-justify-xs-space-between MuiGrid-grid-xs-11 MuiGrid-grid-lg-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-md-12 MuiGrid-grid-lg-3"
      >
        <mock-typography
          classname="Surveys-title-3"
        >
          Survey History
        </mock-typography>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-md-12 MuiGrid-grid-lg-5"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <mock-typography
            classname="PointSelector-subTitle-11"
            variant="h6"
          >
            Survey Point:
          </mock-typography>
        </div>
        <div
          class="MuiGrid-root PointSelector-selectorWrapper-12 MuiGrid-item"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline PointSelector-selector-13"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="survey-point survey-point"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                  id="survey-point"
                  role="button"
                  tabindex="0"
                >
                  All
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  name="survey-point"
                  tabindex="-1"
                  value="All"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <mock-menu
                  anchorel="[object HTMLDivElement]"
                  id="menu-survey-point"
                  menulistprops="[object Object]"
                  open="false"
                  paperprops="[object Object]"
                >
                  <mock-menuitem
                    aria-selected="true"
                    data-value="All"
                    role="option"
                    selected="true"
                  >
                    <mock-typography
                      classname="PointSelector-menuItem-15"
                      variant="h6"
                    >
                      All
                    </mock-typography>
                  </mock-menuitem>
                </mock-menu>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-md-12 MuiGrid-grid-lg-4"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <mock-typography
            classname="Surveys-subTitle-4"
            variant="h6"
          >
            Observation:
          </mock-typography>
        </div>
        <div
          class="MuiGrid-root Surveys-selectorWrapper-5 MuiGrid-item"
        >
          <div
            class="MuiFormControl-root Surveys-formControl-6"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline Surveys-selectedItem-7 MuiInputBase-formControl MuiInput-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="survey-observation survey-observation"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input Surveys-textField-9"
                id="survey-observation"
                role="button"
                tabindex="0"
              >
                <mock-typography
                  classname="Surveys-menuItem-8"
                  variant="h6"
                >
                  Any
                </mock-typography>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                name="survey-observation"
                tabindex="-1"
                value="any"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <mock-menu
                anchorel="[object HTMLDivElement]"
                id="menu-survey-observation"
                menulistprops="[object Object]"
                open="false"
                paperprops="[object Object]"
              >
                <mock-menuitem
                  aria-selected="true"
                  data-value="any"
                  role="option"
                  selected="true"
                >
                  <mock-typography
                    classname="Surveys-menuItem-8"
                    variant="h6"
                  >
                    Any
                  </mock-typography>
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-8"
                  data-value="healthy"
                  role="option"
                  selected="false"
                  title="Appears healthy"
                >
                  Appears healthy
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-8"
                  data-value="possible-disease"
                  role="option"
                  selected="false"
                  title="Showing possible signs of disease"
                >
                  Showing possible signs of disease
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-8"
                  data-value="evident-disease"
                  role="option"
                  selected="false"
                  title="Signs of disease are evident"
                >
                  Signs of disease are evident
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-8"
                  data-value="mortality"
                  role="option"
                  selected="false"
                  title="Signs of mortality are evident"
                >
                  Signs of mortality are evident
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-8"
                  data-value="environmental"
                  role="option"
                  selected="false"
                  title="Signs of environmental disturbance are evident (e.g.,storm damage)"
                >
                  Signs of environmental disturbance are evident (e.g.,storm damage)
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-8"
                  data-value="anthropogenic"
                  role="option"
                  selected="false"
                  title="Signs of anthropogenic disturbance are evident (e.g.,fishing gear, trampling)"
                >
                  Signs of anthropogenic disturbance are evident (e.g.,fishing gear, trampling)
                </mock-menuitem>
              </mock-menu>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-11 MuiGrid-grid-lg-12"
    >
      <div
        class="SurveyTimeline-root-28"
      >
        <mock-hidden
          mddown="true"
        >
          <ul
            class="MuiTimeline-root MuiTimeline-alignLeft"
          >
            <li
              class="MuiTimelineItem-root MuiTimelineItem-alignLeft SurveyTimeline-timelineItem-30"
            >
              <div
                class="MuiTimelineOppositeContent-root MuiTimelineItem-oppositeContent SurveyTimeline-timelineOppositeContent-31"
              >
                <mock-typography
                  classname="SurveyTimeline-dates-21"
                  variant="h6"
                >
                  09/10/2020
                </mock-typography>
              </div>
              <div
                class="MuiTimelineSeparator-root"
              >
                <hr
                  class="SurveyTimeline-connector-29"
                />
                <span
                  class="MuiTimelineDot-root SurveyTimeline-dot-32 MuiTimelineDot-outlinedGrey"
                />
                <hr
                  class="SurveyTimeline-connector-29"
                />
              </div>
              <div
                class="MuiTimelineContent-root MuiTimelineItem-content SurveyTimeline-surveyCardWrapper-23"
              >
                <div
                  class="MuiPaper-root SurveyCard-surveyCard-39 MuiPaper-elevation0 MuiPaper-rounded"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
                    style="height: 100%;"
                  >
                    <div
                      class="MuiGrid-root SurveyCard-cardImageWrapper-43 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
                    >
                      <a
                        href="/reefs/1/survey_details/46"
                      >
                        <div
                          class="MuiCardMedia-root SurveyCard-cardImage-40"
                        />
                      </a>
                    </div>
                    <div
                      class="MuiGrid-root SurveyCard-infoWrapper-44 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
                    >
                      <div
                        class="MuiGrid-root SurveyCard-info-49 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
                        >
                          <mock-typography
                            classname="SurveyCard-cardFields-41"
                            variant="h6"
                          >
                            User:
                          </mock-typography>
                          <mock-typography
                            classname="SurveyCard-cardValues-42 SurveyCard-valuesWithMargin-47"
                            variant="h6"
                          >
                            Joe Doe
                          </mock-typography>
                        </div>
                        <div
                          class="MuiGrid-root SurveyCard-commentsWrapper-45 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-flex-start MuiGrid-grid-xs-12"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                            style="height: 80%;"
                          >
                            <mock-typography
                              classname="SurveyCard-cardFields-41"
                              variant="h6"
                            >
                              Comments:
                            </mock-typography>
                            <mock-typography
                              classname="SurveyCard-cardValues-42 SurveyCard-comments-46"
                              variant="h6"
                            >
                              No comments
                            </mock-typography>
                          </div>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
                        >
                          <mock-typography
                            classname="SurveyCard-cardFields-41"
                            variant="h6"
                          >
                            Temp:
                          </mock-typography>
                          <mock-typography
                            classname="SurveyCard-cardValues-42 SurveyCard-valuesWithMargin-47"
                            variant="h6"
                          >
                            10.0 °C
                          </mock-typography>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between MuiGrid-grid-xs-12"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
                          >
                            <a
                              href="/reefs/1/survey_details/46"
                              style="text-decoration: none;"
                            >
                              <mock-button
                                color="primary"
                                size="small"
                                variant="outlined"
                              >
                                VIEW DETAILS
                              </mock-button>
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </mock-hidden>
        <mock-hidden
          lgup="true"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root SurveyTimeline-surveyWrapper-33 MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root SurveyTimeline-dateWrapper-20 MuiGrid-item MuiGrid-grid-xs-11"
              >
                <mock-typography
                  classname="SurveyTimeline-dates-21"
                  variant="h6"
                >
                  09/10/2020
                </mock-typography>
              </div>
              <div
                class="MuiGrid-root SurveyTimeline-surveyCardWrapper-23 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiPaper-root SurveyCard-surveyCard-39 MuiPaper-elevation0 MuiPaper-rounded"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
                    style="height: 100%;"
                  >
                    <div
                      class="MuiGrid-root SurveyCard-cardImageWrapper-43 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
                    >
                      <a
                        href="/reefs/1/survey_details/46"
                      >
                        <div
                          class="MuiCardMedia-root SurveyCard-cardImage-40"
                        />
                      </a>
                    </div>
                    <div
                      class="MuiGrid-root SurveyCard-infoWrapper-44 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
                    >
                      <div
                        class="MuiGrid-root SurveyCard-info-49 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
                        >
                          <mock-typography
                            classname="SurveyCard-cardFields-41"
                            variant="h6"
                          >
                            User:
                          </mock-typography>
                          <mock-typography
                            classname="SurveyCard-cardValues-42 SurveyCard-valuesWithMargin-47"
                            variant="h6"
                          >
                            Joe Doe
                          </mock-typography>
                        </div>
                        <div
                          class="MuiGrid-root SurveyCard-commentsWrapper-45 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-flex-start MuiGrid-grid-xs-12"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                            style="height: 80%;"
                          >
                            <mock-typography
                              classname="SurveyCard-cardFields-41"
                              variant="h6"
                            >
                              Comments:
                            </mock-typography>
                            <mock-typography
                              classname="SurveyCard-cardValues-42 SurveyCard-comments-46"
                              variant="h6"
                            >
                              No comments
                            </mock-typography>
                          </div>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
                        >
                          <mock-typography
                            classname="SurveyCard-cardFields-41"
                            variant="h6"
                          >
                            Temp:
                          </mock-typography>
                          <mock-typography
                            classname="SurveyCard-cardValues-42 SurveyCard-valuesWithMargin-47"
                            variant="h6"
                          >
                            10.0 °C
                          </mock-typography>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between MuiGrid-grid-xs-12"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
                          >
                            <a
                              href="/reefs/1/survey_details/46"
                              style="text-decoration: none;"
                            >
                              <mock-button
                                color="primary"
                                size="small"
                                variant="outlined"
                              >
                                VIEW DETAILS
                              </mock-button>
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </mock-hidden>
      </div>
    </div>
  </div>
</div>
`;
